soccer/src/soccer/strategy/agent/position/position.cpp

```diff
@@ -71,22 +71,167 @@ bool Position::assert_world_state_valid() {
     return true;
 }
 
-communication::PosAgentRequestWrapper Position::send_communication_request() {
-    return communication_request_;
+std::optional<communication::PosAgentRequestWrapper> Position::send_communication_request() {
+    if (communication_request_ != std::nullopt) {
+        auto saved_comm_req = communication_request_;
+        communication_request_ = std::nullopt;
+        return saved_comm_req;
+    }
+    return std::nullopt;
+}
+
+void Position::receive_communication_response(communication::AgentPosResponseWrapper response) {
```

Comment:
also, upon doing some deep diving, I can't figure out why you chose to bundle fields in a struct (AgentPosResponseWrapper) instead of making them params to this method. why did you do that?

Reply:
The main reason there are the wrappers instead of method parameters is for the relaying of requests from the position to the agent and the relaying of responses from the agent to the position.

For the position to agent request case, although there are only 4 parameters (the request, the target agents, broadcast, and urgent), 3 of the 4 fields are not required in every case (i.e. target agents, broadcast, and urgent).  When I was writing everything I also forgot that C++ has the ability to give default parameters to variables (Rust does not) so I figured it would be easier to wrap all of the fields together instead of writing a different method for each case. 

For the agent the position response case, however, there is a lot of information to relay (some of which may be not always be used) so it felt like it would be annoying for a user to look at a function with 7 arguments of which only 1 is always used.  This wrapper is also used by the agent action client to handle the aggregation of responses to a request before relaying them to the current position and it made sense to do this in a singular aggregated data type.

Reply:
okay, fine by me then